presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RewriteAggregationIfToFilter.java

```diff
@@ -56,13 +60,16 @@
  * AGG(IF(condition, expr))
  * to
  * AGG(IF(condition, expr)) FILTER (WHERE condition).
+ * or
+ * AGG(expr) FILTER (WHERE condition).
  * <p>
- * The latter plan is more efficient because:
+ * The rewritten plan is more efficient because:
  * 1. The filter can be pushed down to the scan node.
  * 2. The rows not matching the condition are not aggregated.
  * <p>
- * This rule doesn't unwrap the IF expression in the aggregate, because the true branch might return errors for rows not matching the filters. For example:
+ * Note that unwrapping the IF expression in the aggregate might cause issues if the true branch return errors for rows not matching the filters. For example:
  * 'IF(CARDINALITY(array) > 0, array[1]))'
+ * Session property AGGREGATION_IF_TO_FILTER_REWRITE_UNWRAP_IF_ENABLED and canUnwrapIf() control whether to enable IF unwrapping.
  */
 public class RewriteAggregationIfToFilter
```

Comment:
For the `isEnabled` function, you could use the cardinal of an enum to decide if we should apply this rule or not.

Reply:
Done.